checkov/terraform/module_loading/registry.py

```diff
@@ -42,15 +43,25 @@ def load(
         """
         if source is None:
             return None
-
         if module_address is None:
             module_address = f'{source}:{source_version}'
         if module_address in self.module_content_cache:
             logging.debug(f'Used the cache for module {module_address}')
             return self.module_content_cache[module_address]
-        else:
-            logging.debug(f'Cache miss for {module_address}')
 
+        # If we have tf managed modules, we likely have whatever :latest is in the cache
+        if env_vars_config.CHECKOV_EXPERIMENTAL_TERRAFORM_MANAGED_MODULES:
+            if source_version == 'latest':
+                vers = sorted([k.rsplit(':', 1)[1] for k in self.module_content_cache.keys() if k.startswith(f'{source}:')])
+                if vers:
+                    logging.debug(f'Used the cache for module {module_address}')
+                    return self.module_content_cache[f'{source}:{vers[-1]}']
+                vers = sorted([k.rsplit(':', 1)[1] for k in self.module_content_cache.keys() if k.startswith(f'registry.terraform.io/{source}:')])
+                if vers:
+                    logging.debug(f'Used the cache for module (from tf registry) {module_address}')
+                    return self.module_content_cache[f'registry.terraform.io/{source}:{vers[-1]}']
```

Comment:
sorting doesn't give any benefit and just costs time. also iterating two times over the same collection seems rather wasteful and can be done by a simple loop and checking for either the normal case or the registry prefix. accessing the module cache with the concatenated key adds an additional lookup (plus the string concatenation), you can can iterate over the `.items()` and have the value right next to the key.

Reply:
Even better, just keep a separate lookaside.